src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs

```diff
@@ -102,7 +102,9 @@ protected string GetTestFileName(int? index = null, [CallerMemberName] string me
                 if (excessLength < memberName.Length + "...".Length)
                 {
                     // Take a chunk out of the middle as perhaps it's the least interesting part of the name
-                    memberName = memberName.Substring(0, memberName.Length / 2 - excessLength / 2) + "..." + memberName.Substring(memberName.Length / 2 + excessLength / 2);
+                    int halfMemberNameLength = (int)Math.Floor((double)memberName.Length / 2);
+                    int halfExcessLength = (int)Math.Floor((double)excessLength / 2);
```

Comment:
This should probably be Ceil not Floor, and then we don't need the +1 on the math.